sklbench/utils/bench_case.py

```diff
@@ -0,0 +1,124 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+from copy import deepcopy
+from typing import Any, List, Union
+
+from .custom_types import BenchCase, JsonTypesUnion
+
+
+def set_bench_case_value(
+    bench_case: BenchCase, param_name: Union[List[str], str], value: JsonTypesUnion
+):
+    if isinstance(param_name, str):
+        keys_chain = param_name.split(":")
+    else:
+        keys_chain = param_name
+    # deep dive into bench case
+    local_value = bench_case
+    for prev_key in keys_chain[:-1]:
+        if prev_key not in local_value:
+            local_value[prev_key] = dict()
+        local_value = local_value[prev_key]
+    local_value[keys_chain[-1]] = value  # type: ignore
+
+
+def get_bench_case_value(
+    bench_case: BenchCase,
+    param_name: Union[List[str], str],
+    default_value: JsonTypesUnion = None,
+) -> Any:
+    if isinstance(param_name, str):
+        keys_chain = param_name.split(":")
+    else:
+        keys_chain = param_name
+    # deep dive into bench case
+    local_value = bench_case
+    for prev_key in keys_chain:
+        if prev_key not in local_value:
+            return default_value
+        local_value = local_value[prev_key]
+    return deepcopy(local_value)
+
+
+def get_bench_case_values(
+    bench_case: BenchCase,
+    param_names: Union[List[List[str]], List[str]],
+    default_value: JsonTypesUnion = None,
+) -> List[Any]:
+    return list(
+        get_bench_case_value(bench_case, param_name, default_value)
+        for param_name in param_names
+    )
+
+
+def get_first_of_bench_case_values(
+    bench_case: BenchCase, param_names: Union[List[List[str]], List[str]]
+) -> JsonTypesUnion:
+    values = get_bench_case_values(bench_case, param_names, None)
+    values = list(filter(lambda x: x is not None, values))
+    if len(values) == 0:
+        raise ValueError(f"Unable to find any of values: {param_names}.")
+    else:
+        return values[0]
+
+
+def get_data_name(bench_case: BenchCase, shortened: bool = False) -> str:
+    # check if unique dataset name is specified directly
+    dataset = get_bench_case_value(bench_case, "data:dataset")
+    if dataset is not None:
+        return dataset
+    # check source of data
+    source = get_bench_case_value(bench_case, "data:source")
+    # generate kwargs postfixes for data filename
+    postfixes = dict()
+    for kwargs_type in ["generation", "dataset"]:
+        postfix = ""
+        for key, value in get_bench_case_value(
+            bench_case, f"data:{kwargs_type}_kwargs", dict()
+        ).items():
+            postfix += f"_{key}_{value}"
```

Comment:
```suggestion
            postfix += f"_{key}_{value}".replace('.', '_')
```
avoids creating dataset from float args with extra periods before extension, can use underscore or something else

Reply:
Please address

Reply:
Extra underscores will make file name confusing.

Reply:
There needs to be some alternative because we are currently adding in period into file name, which leads to issues loading from cache

Reply:
There needs to be some alternative because we are currently adding in period into file name, which leads to issues loading from cache

Reply:
Example: have noise param in regression synthetic data generation, leads to make_regression_n_samples_255000_n_features_10000_noise_1.25_x.npz creation in the cache, then when trying to load this data it fails because of the extra period. If an underscore is not preferred then something else perhaps.

Reply:
I don't observe such issue as similar file names are loaded correctly from cache on my side.